bin/opd8n/src/cmd/from_l1.rs

```diff
@@ -114,55 +194,105 @@ impl FromL1 {
         Ok(())
     }
 
-    /// Derives the L2 output roots from the provided L1 block range.
-    pub async fn derive(&self) -> Result<Vec<B256>> {
-        panic!("l2 output root derivation is not supported");
+    /// Gets the L2 starting block number.
+    /// Returns the genesis L2 block number if the start block is less than the genesis block number.
+    pub fn start_block(&self, cfg: &RollupConfig) -> u64 {
+        if self.start_block < cfg.genesis.l2.number {
+            cfg.genesis.l2.number
+        } else if self.start_block != 0 {
+            self.start_block - 1
+        } else {
+            self.start_block
+        }
     }
```

Comment:
Consider using match statements.

```suggestion
    pub fn start_block(&self, cfg: &RollupConfig) -> u64 {
        match self.start_block {
            block if block < cfg.genesis.l2.number => cfg.genesis.l2.number,
            block if block != 0 => sb - 1,
            _ => self.start_block,
        }
    }
```